### PR TITLE
Feat: allow show() to take optional detailsPromise

### LINKED ISSUE #645: Add way to update `total` and `displayItems` after `.show()` but before user interaction
This is related to but different from #639.

Some users have reported to us that they may not know the total or line items at the time of attempting to open the payment sheet with `show()` -- there may be asynchronous server-based logic yet to be resolved.

Apple Pay provides an unrelated API that users can repurpose to work around this: if you listen for [onpaymentmethodselected](https://developer.apple.com/documentation/applepayjs/applepaysession/1778013-onpaymentmethodselected), you can get a loading state when the sheet opens and then asynchronously update the total and return control to the sheet. We have at least one user that has been relying on this behavior with Apple Pay.

Is it worth adding a way to signal that the browser should show a spinner or equivalent and wait to allow end user interaction until an update? The alternative is for Payment Request implementors to block the end-user action (not great) or show a manually-created loading modal (probably also not great).

### PR DESCRIPTION
**Pull request by marcoscaceres**
closes #645 

 * [x] tests - [Part 1 - rejects](https://github.com/w3c/web-platform-tests/pull/9617), [Part 2 - resolves](https://github.com/w3c/web-platform-tests/pull/9617)
 * [x] [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/PaymentRequest/show)

Implementer issues filed:
   * [x] [Safari](https://bugs.webkit.org/show_bug.cgi?id=182538) 
   * [x] [Chrome](https://bugs.chromium.org/p/chromium/issues/detail?id=817073)
   * [x] Edge - ack from @adrianba
   * [x] [Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1441709)

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/pull/672.html" title="Last updated on Feb 28, 2018, 4:47 AM GMT (6255cc0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/payment-request/672/f5096c6...6255cc0.html" title="Last updated on Feb 28, 2018, 4:47 AM GMT (6255cc0)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -932,12 +933,34 @@ <h2>
...
+          interact with the <var>handlers</var>. The user agent SHOULD
+          prioritize the preference of the user when presenting payment
+          methods.
+          </li>
+          <li>If <var>detailsPromise</var> was passed, then:
+            <ol>
+              <li>Set <var>request</var>.<a>[[\updating]]</a> to true.
+              </li>
+              <li>Disable the user interface user interface that will allow the
+              user to interact with the <var>handlers</var> and run the
```
Comment (domenic) [inline index.html]: Make this two steps?

**Review by domenic — APPROVED** [review]
Elsewhere it says

>  SHOULD re-enable user interface elements that might have been disabled in the steps above if appropriate. 

this should probably say "prior to running this algorithm" instead of "in the steps above"

LGTM with nits

**Comment by marcoscaceres** [comment]
Thanks again for the review, @domenic!

**Comment by marcoscaceres** [comment]
Part 1 of tests https://github.com/w3c/web-platform-tests/pull/9617

**Comment by marcoscaceres** [comment]
Tests are complete, awaiting review on: [Part 1 - rejects](https://github.com/w3c/web-platform-tests/pull/9617). 

@aestes, not sure if it helps you much, but I added ApplePay as a payment method identifier to the tests. Let me know if there is anything I can do in web platform tests to make it easier to test in WebKit/Safari.

**Comment by marcoscaceres** [comment]
I've added MDN documentation for this.

**Comment by marcoscaceres** [comment]
@ianbjacobs, could you kindly please request the specification be republished.

**Comment by ianbjacobs** [comment]
@marcoscaceres, request sent today for 1 March publication.

**Pull request merged**